contracts/src/bridge/SequencerInbox.sol

```diff
@@ -133,6 +142,10 @@ contract SequencerInbox is DelegateCallAware, GasRefundEnabled, ISequencerInbox
         if (msg.sender != tx.origin) revert NotOrigin();
         if (!isBatchPoster[msg.sender]) revert NotBatchPoster();
         if (inboxAccs.length != sequenceNumber) revert BadSequencerNumber();
+        bytes32 dasKeysetHash = dasKeysetHashFromBatchData(data);
+        if (dasKeysetHash != bytes32(0)) {
+            if (!isValidKeysetHash[dasKeysetHash]) revert InvalidDASKeyset(dasKeysetHash);
```

Comment:
This check needs to happen in `formDataHash` so that it applies to the non-origin variant